flask_chat/views.py

```diff
@@ -0,0 +1,414 @@
+
+    # Check whether user exists in the database
+    if not user_service.get_user_info(username=username):
+        flash("Invalid username or password.")
+        return render_template("login.html", form=form)
+
+    [user_data] = user_service.get_user_info(username=username)
+    if not user_data:
+        logger.error("Unable to load user info.")
+        return render_template(
+            "error.html",
+            message="Unable to process info about your identity. Pleasy try later.",
+        )
+
+    # Here goes verifying password hashes
+    if not user_data.verify_password(password):
+        flash("Invalid username or password.")
+        logger.info("Login failed")
+        return render_template("login.html", form=form)
+
+    # Creating session token for user, considering it as successful login,
+    # redirecting to main page
+    response = redirect("/")
+    access_token = create_access_token(identity=str(user_data.user_id))
+    set_access_cookies(response, access_token)
+
+    logger.info("Login successful.")
+
+    return response
+
+
+@views_bp.route("/logout")
+@jwt_required()
+def logout():
+    """Logout function"""
+
+    log_request()
+    response = redirect("/login")
+    unset_jwt_cookies(response)
+
+    logger.info("User logged out")
+
+    return response
+
+
+@views_bp.route("/myprofile", methods=["GET", "POST"])
+@jwt_required()
+def profile():
+    """Display profile page"""
+
+    log_request()
+
+    user_id = get_jwt_identity()
+
+    # Retrieve info from database about user
+    user_data = user_service.get_user_by_id(user_id)
+    if not user_data:
+        logger.error("Unable to load user info.")
+        return render_template(
+            "error.html",
+            message="Unable to process info about your identity. Pleasy try later.",
+        )
+
+    user_data = user_data.to_json()
+    current_user = user_data.get("username")
+    logger.debug("Current user: %s", current_user)
+
+    # Get csrf token to put in rendered form
+    csrf_token = request.cookies.get("csrf_access_token")
+
+    # Edit profile form handle
+    form = EditProfileForm(request.form)
+
+    if request.method != "POST":
+
+        # Placeholder values from database to display in profile
+        for field in form:
+            if field.name == "csrf_token":
+                logger.debug("CSRF token is set")
+                field.data = csrf_token
+            elif user_data.get(field.name):
+                field.data = user_data.get(field.name)
+            # If the field from user's document is empty
+            else:
+                field.data = ""
+
+        return render_template(
+            "profile.html",
+            form=form,
+            current_user=current_user,
+            csrf_token=csrf_token,
+            data=user_data,
+            web_name=WEBSITE_NAME,
+        )
+
+    # Check if request contains data about file uploaded
+    if request.files["file"].read() != b"":
```

Comment:
Please, delete redundant code with seek() and double read of file. Just read file into variable, than do if

Reply:
Fixed.